src/CoreWCF.Primitives/src/CoreWCF/Channels/ServiceChannel.cs

```diff
@@ -764,7 +770,7 @@ private TimeoutException GetOpenTimeoutException(TimeSpan timeout)
             }
         }
 
-        internal void HandleReceiveComplete(RequestContext context)
+        internal Task HandleReceiveComplete(RequestContext context)
         {
```

Comment:
Can you rename the method to have the Async suffix? Are there other call points which now need to await this method elsewhere in the code?

Reply:
There is only one call site to this method and it's internal. If I rename this, what about all the other methods, e.g. `DecrementActitivity`, `ProcessMessageCleanup` in the `ImmutableDispatchRuntime` etc. Should I rename all of those too, at least the internal ones?

Reply:
If they are async, they should have the Async suffix so rename anything internal which you are touching that is async and doesn't match that name.